src/lib/formValidator.ts

```diff
@@ -0,0 +1,33 @@
+import { IFieldValidator } from "./fieldValidator";
+
+export class FormValidator {
+  private _fields: IFieldValidator[];
+
+  constructor(...fields: IFieldValidator[]) {
+    this._fields = fields;
+  }
+
+  public hasError(): boolean {
+    return this._fields.some((field) => {
+      return field.hasError();
+    });
+  }
+
+  public isValid(): boolean {
+    return !this.hasError();
+  }
+
+  public field(fieldId: string): IFieldValidator {
```

Comment:
Could this type here be:

````typescript
public field(fieldId: keyof Beacon): IFieldValidator {
...
}
````

Or from the `BeaconCacheEntry` to give us type safety that the field is a key of the beacon interface?